### ISSUE TITLE
Add Plugins Property To TSServer Open Request

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#16008]
From https://github.com/Microsoft/vscode/issues/25740

**Problem**
TypeScript server plugins are currently able to tell vscode to send the TypeScript server change events for a set of file types, such as `ng-html` or `.html`. If this is enabled currently, it causes TypeScript to start treating the html file contents as typescript, even if the plugin has not been loaded

**Proposal**
On the `open` request to the `tsserver`, add a `plugins` property to specially mark the opened file as belonging to a plugin. Files with a `plugins` property would not be included in the project automatically but only after the plugin marks the file for inclusion

**Comment by mjbvz** [comment]
Duplicate of #17151
